src/libsyntax_ext/source_util.rs

```diff
@@ -114,20 +112,17 @@ pub fn expand_include_str(cx: &mut ExtCtxt<'_>, sp: Span, tts: &[tokenstream::To
         None => return DummyResult::expr(sp)
     };
     let file = cx.resolve_path(file, sp);
-    match fs::read_to_string(&file) {
-        Ok(src) => {
-            let interned_src = Symbol::intern(&src);
-
-            // Add this input file to the code map to make it available as
-            // dependency information
-            cx.source_map().new_source_file(file.into(), src);
-
-            base::MacEager::expr(cx.expr_str(sp, interned_src))
+    match cx.source_map().load_binary_file(&file) {
```

Comment:
I am not sure whether `include_str` should normalize file contents... This approach does not do normalization, just like the previous version.

Reply:
I'm pretty sure it should.
The content included by `include_str` is still text / source code, if it's a long multi-line string encoded on Windows and stored in a file with UTF-8 BOM, then all those `\r\n`s and BOM shouldn't get into the string for the same reason why any other (non-`include_bytes`) inclusion method filters them away from string literals.

Am I right that https://github.com/rust-lang/rust/pull/62948 applied the normalization to `include_str` as well and crater found no regression?
This should give as the right to make the change (in this PR or in #62948, not much difference).